03-task-manager/starter/middleware/not-found.js

```diff
@@ -0,0 +1,11 @@
+const { CustomAPIError } = require('../errors/custom-error');
+
+const errorHandlerMiddleware = (err, req, res, next) => {
+  if (err instanceof CustomAPIError) {
+    return res.status(err.statusCode).json({ msg: err.message });
+  }
+
+  return res.status(500).json({ msg: 'Something went wrong!' });
+};
+
+module.exports = errorHandlerMiddleware;
```

Comment:
This `not-found.js` is meant to be used as a catch-all / backup for when a user navigates to a route that isn't defined in your app. Usually, `500` status codes are used when we want to tell the user "Sorry something went wrong on our end". And `4xx` series status codes are used to tell the user "Hey sorry you did something wrong". Specifically, a `404` is the common HTTP status code to say "This URL does not exist". This blog post has some common ones: [https://blog.webdevsimplified.com/2022-12/http-status-codes/](https://blog.webdevsimplified.com/2022-12/http-status-codes/)

In this case, we're not catching an error thrown inside our code, so we don't need to handle `CusotmAPIError`. Rather we're just using this middleware to set the response to \`404.

That's why this middleware has to be declared _after_ we declare all the routes in `app.js` (because if we had `app.use(notFound);` before `app.use('/api/v1/tasks', tasks);` then the notFound middleware would run before Express even checks any of the routes for matches. And the user would get a 404 even on pages that should work).

Here's what this code should look like:

```suggestion
const notFound = (req, res) => res.status(404).send('Route does not exist');

module.exports = notFound;
```

Reply:
@hammaazarok same with this? Then you'll be able to merge this PR when you're ready and the fix should propogate (hopefully without needing to do too much advanced git-fu 😅 ) to the later PRs

Reply:
I hope so 👍